src/cli/self_update.rs

```diff
@@ -61,20 +57,81 @@ fn default_archive_name() -> Option<String> {
     }
 }
 
-pub async fn execute(args: Args) -> miette::Result<()> {
-    // Retrieve the target version information from github.
-    let target_version_json = match retrieve_target_version(&args.version).await {
-        Ok(target_version_json) => target_version_json,
-        Err(err) => match args.version {
-            Some(version) => {
-                miette::bail!("The version you specified is not available: {}", version)
+async fn latest_version(base_url: Url) -> miette::Result<String> {
+    // Uses the public Github Releases /latest endpoint to get the latest tag from the URL
+    // If base_url doesn't offer redirects (e.g. static site), then /latest
+    // must be a file whose contents are the latest tag name.
+    let url = base_url.join("latest").into_diagnostic()?;
+    // Create a client with a redirect policy
+    let no_redirect_client = Client::builder()
+        .redirect(Policy::none()) // Prevent automatic redirects
+        .build()
+        .into_diagnostic()?;
+
+    match no_redirect_client
+        .head(url.clone())
+        .header("User-Agent", user_agent())
+        .send()
+        .await
+    {
+        Ok(response) => {
+            if response.status().is_redirection() {
+                match response.headers().get("Location") {
+                    Some(location) => {
+                        let url =
+                            Url::parse(location.to_str().into_diagnostic()?).into_diagnostic()?;
+                        let version = url
+                            .path_segments()
+                            .ok_or_else(|| {
+                                miette::miette!("Could not get segments from Location header")
+                            })?
+                            .last()
+                            .ok_or_else(|| {
+                                miette::miette!("Could not get version from Location header")
+                            })?;
+                        Ok((version.strip_prefix("v").unwrap_or(&version)).to_string())
```

Comment:
I'd say, that we should error out if the version string doesn't start with a "v"

Reply:
Something like [2a3e0a5](https://github.com/prefix-dev/pixi/pull/2989/commits/2a3e0a5cc711e752830a107b0b20e3f6129af61f)?

Reply:
Kind of. You now don't strip the "v" anymore, or am I missing something? :)

Reply:
Now we are back to doing it heh, sorry for the noise!